src/improvements/tasks/OPCMBaseTask.sol

```diff
@@ -57,10 +59,15 @@ abstract contract OPCMBaseTask is MultisigTask {
         data = abi.encodeWithSignature("aggregate3((address,bool,bytes)[])", calls);
     }
 
-    function validate() public view override {
+    function validate(VmSafe.AccountAccess[] memory accesses) public override {
         (address[] memory targets,,) = getTaskActions();
         require(targets.length == 1 && targets[0] == opcm(), "OPCMBaseTask: only OPCM is allowed as target");
-        super.validate();
+        super.validate(accesses);
+        require(
+            _stateInfos[parentMultisig].length == 1,
+            "OPCMBaseTask: only nonce should be updated on upgrade controller multisig"
+        );
+        require(_stateInfos[opcm()].length == 0, "OPCMBaseTask: Storage writes are not allowed on OPCM");
```

Comment:
Once https://github.com/ethereum-optimism/superchain-ops/pull/616 is merged we should use that library instead of relying on the stateful read of `_stateInfos` here